pkg/operator/ceph/object/controller_test.go

```diff
@@ -655,3 +662,250 @@ func TestCephObjectStoreControllerMultisite(t *testing.T) {
+						}
+						if args[2] == "set" {
+							return "", nil // success
+						}
+					}
+				}
+				if args[0] == "config" {
+					if args[1] == "get" && args[3] == "rgw_rados_pool_pg_num_min" {
+						// ceph config get mon. rgw_rados_pool_pg_num_min
+						return "8", nil
+					}
+					if args[1] == "set" && args[2] == "client.rgw.my.store.a" {
+						return "", nil // success
+					}
+				}
+				if args[0] == "auth" && args[1] == "get-or-create-key" {
+					return `{"key":"mysecurekey"}`, nil
+				}
+				if args[0] == "dashboard" && args[1] == "get-rgw-api-access-key" {
+					return access_key, nil
+				}
+				panic(fmt.Sprintf("unhandled command %s %v", command, args))
+			},
+			MockExecuteCommandWithTimeout: func(timeout time.Duration, command string, args ...string) (string, error) {
+				t.Logf("Command: %s %v", command, args)
+				if command == "radosgw-admin" {
+					if args[0] == "user" {
+						return userCreateJSON, nil
+					}
+					if args[0] == "realm" && args[1] == "get" {
+						return rgwAdminRealmGetOutput, nil
+					}
+					if args[0] == "zonegroup" && args[1] == "get" {
+						return rgwAdminZoneGroupGetOutput, nil
+					}
+					if args[0] == "zone" && args[1] == "get" {
+						return rgwAdminZoneGetOutput, nil
+					}
+				}
+				panic(fmt.Sprintf("unhandled command %s %v", command, args))
+			},
+		}
+
+		objStore := &cephv1.CephObjectStore{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "CephObjectStore",
+				APIVersion: "ceph.rook.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      storeName,
+				Namespace: ns,
+			},
+			Spec: cephv1.ObjectStoreSpec{
+				Gateway: cephv1.GatewaySpec{
+					Port: 80,
+				},
+			},
+		}
+
+		controllerRuntimeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(objStore).Build()
+
+		reconciler := ReconcileCephObjectStore{
+			client: controllerRuntimeClient,
+			// bktclient           bktclient.Interface
+			scheme: scheme,
+			// context             *clusterd.Context
+			context: &clusterd.Context{
+				Clientset: k8sfake.NewSimpleClientset(),
+				Executor:  executor,
+			},
+			clusterSpec: &cephv1.ClusterSpec{
+				DataDirHostPath: "/var/lib/rook",
+			},
+			clusterInfo: &cephclient.ClusterInfo{
+				Namespace: ns,
+			},
+			objectStoreChannels: map[string]*objectStoreHealth{
+				storeName: {
+					monitoringRunning: false,
+				},
+			},
+			// recorder            *k8sutil.EventReporter
+		}
+
+		origHTTPClientFunc := genObjectStoreHTTPClientFunc
+		genObjectStoreHTTPClientFunc = func(objContext *Context, spec *cephv1.ObjectStoreSpec) (client *http.Client, tlsCert []byte, err error) {
+			return nil, []byte{}, errors.New("induced error creating admin ops API connection")
+		}
+		defer func() { genObjectStoreHTTPClientFunc = origHTTPClientFunc }()
+
+		_, err := reconciler.reconcileCreateObjectStore(objStore, nsName, cluster)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "failed to start rgw health checker")
```

Comment:
I think this could fit in `TestCephObjectStoreController` which has been rewritten in https://github.com/rook/rook/pull/8607 for more clarity. This would avoid duplicating boilet plates too. We would test that the `Requeue` in the response is `true`.

Should this wait for https://github.com/rook/rook/pull/8607?

Reply:
I think this can wait for 8607 for sure. I'm happy to add this as part of existing boilerplate. Maybe we could merge #8607 earlier to allow #5313 to use the boilerplate for unit testing?

I definitely agree that other controllers need tests like this. I just wanted to add the minimal necessary test for this regression here and leave other work for later.